ui/dashboards/src/components/EditDashboardButton/EditDashboardButton.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Button } from '@mui/material';
+import PencilIcon from 'mdi-material-ui/PencilOutline';
+
+export interface EditDashboardButtonProps {
+  /**
+   * The label used inside the button.
+   */
+  label?: string;
+
+  /**
+   * Handler that puts the dashboard into editing mode.
+   */
+  onClick: () => void;
+}
+
+export const EditDashboardButton = ({ label = 'Edit', onClick }: EditDashboardButtonProps) => {
+  return (
+    <Button
+      onClick={onClick}
+      startIcon={<PencilIcon />}
+      variant="outlined"
+      color="secondary"
```

Comment:
Not making these customizable to start because we do not have a use case for it.